### PR TITLE
8340229: Improve opening sentence of FileInputStream constructor specification

### PR DESCRIPTION
**Pull request by bplb**
Improve the first sentences of the three `FileInputStream` constructors, in particular removing the term "connection."

### CODE REVIEW

File: src/java.base/share/classes/java/io/FileInputStream.java [hunk 1]
```diff
@@ -115,10 +114,8 @@ public FileInputStream(String name) throws FileNotFoundException {
     }
 
     /**
-     * Creates a {@code FileInputStream} by
-     * opening a connection to an actual file,
-     * the file named by the {@code File}
-     * object {@code file} in the file system.
+     * Creates a {@code FileInputStream} to read from an existing file
+     * named by the {@code File} object {@code file}.
```
Comment (jaikiran) [inline src/java.base/share/classes/java/io/FileInputStream.java:118]: Hello Brian, should be perhaps reword this as "... to read from an existing file represented by the {@code file}." This would almost match what we state for the constructor which takes the `FileDescriptor` instance.

**Comment by jaikiran** [comment]
Do you think, as part of this PR, we should also revisit some of the javadoc in the `java.io.FileOutputStream` class which too has several mentions of "connection"?

File: src/java.base/share/classes/java/io/FileInputStream.java [hunk 2]
```diff
@@ -165,9 +162,8 @@ public FileInputStream(File file) throws FileNotFoundException {
     }
 
     /**
-     * Creates a {@code FileInputStream} by using the file descriptor
-     * {@code fdObj}, which represents an existing connection to an
-     * actual file in the file system.
+     * Creates a {@code FileInputStream} to read from an existing file
+     * represented by the {@code FileDescriptor} object {@code fdObj}.
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/io/FileInputStream.java:166]: I don't know if you meant to touch this constructor or not but I think it's okay to use "connected" here as this is creating a FIS to read bytes from whatever the file descriptor is connected to.

Comment (bplb) [reply src/java.base/share/classes/java/io/FileInputStream.java:166]: I have reverted it locally and that will be in the next commit.

File: src/java.base/share/classes/java/io/FileInputStream.java (hunk 1 above)
Comment (bplb) [reply src/java.base/share/classes/java/io/FileInputStream.java:118]: Updated for the next commit.

**Comment by bplb** [comment]
> Do you think, as part of this PR, we should also revisit some of the javadoc in the `java.io.FileOutputStream` class which too has several mentions of "connection"?

Are you thinking of this sentence 

```A new FileDescriptor object is created to represent this file connection.```

which appears for the four constructors which do not have a `FileDescriptor` parameter? Those four sentences could probably be deleted.

**Comment by jaikiran** [comment]
> > Do you think, as part of this PR, we should also revisit some of the javadoc in the `java.io.FileOutputStream` class which too has several mentions of "connection"?
> 
> Are you thinking of this sentence
> 
> `A new FileDescriptor object is created to represent this file connection.`
> 
> which appears for the four constructors which do not have a `FileDescriptor` parameter? Those four sentences could probably be deleted.

Yes those were the occurences that I was curious about. Having said that, I now see that we use similar sentence even in the 2 constructors of FileInputStream and we aren't proposing to update those in this PR. So I think it should be OK to leave it in its current form in FileOutputStream.

File: src/java.base/share/classes/java/io/FileInputStream.java (hunk 1 above)
Comment (AlanBateman) [reply src/java.base/share/classes/java/io/FileInputStream.java:118]: Did you push that commit? I don't see the update, the PR only has 1 commit.

Comment (bplb) [reply src/java.base/share/classes/java/io/FileInputStream.java]: I did just now.

File: src/java.base/share/classes/java/io/FileInputStream.java [hunk 3]
```diff
@@ -115,10 +114,8 @@ public FileInputStream(String name) throws FileNotFoundException {
     }
 
     /**
-     * Creates a {@code FileInputStream} by
-     * opening a connection to an actual file,
-     * the file named by the {@code File}
-     * object {@code file} in the file system.
+     * Creates a {@code FileInputStream} to read from an existing file
+     * reoresented by the {@code File} object {@code file}.
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/io/FileInputStream.java:118]:
```suggestion
     * represented by the {@code File} object {@code file}.
```

Comment (bplb) [reply src/java.base/share/classes/java/io/FileInputStream.java]: Fixed in c03751d.

**Review by AlanBateman — APPROVED** [review]

**Review by jaikiran — APPROVED** [review]
